ember-server/src/test/scala/org/http4s/ember/server/EmberUnixSocketSuite.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2019 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.ember.server
+
+import org.http4s.ember.client._
+
+import cats.syntax.all._
+import cats.effect._
+import org.http4s._
+import org.http4s.ember.client.EmberClientBuilder
+import fs2.io.net.unixsocket.{UnixSocketAddress, UnixSockets}
+import org.http4s._
+import org.http4s.client.middleware.UnixSocket
+import scala.concurrent.duration._
+
+class EmberUnixSocketSuite extends Http4sSuite {
+
+  test("Should be able to connect to a server created") {
+    def app = HttpApp.liftF(Response[IO](Status.Ok).withEntity("Hello Unix Sockets!").pure[IO])
+    val test = for {
+      id <- Resource.eval(IO(java.util.UUID.randomUUID().toString()))
+      localSocket = UnixSocketAddress("/tmp/" ++ id ++ ".sock")
```

Comment:
Can we use fs2 temporary files method for this?

Reply:
Why? The server creates and deletes its own unix socket by default? The likelihood of a conflict is practically non-existent?

Reply:
Fair point I guess, I don't really understand about these socket files. But I remember when researching temporary files for the fs2.io JS port tricks like this were strongly discouraged in favour of using OS-provided temporary files methods for security reasons. OTOH this is just a test in CI so probably nbd :)

```suggestion
      id <- Resource.eval(cats.effect.std.UUIDGen[IO].randomUUID).map(_.toString)
      localSocket = UnixSocketAddress("/tmp/" ++ id ++ ".sock")
```

Reply:
The JVM has a system property for the temp directory, `java.io.tmpdir`, which would be polite to honor.  It's not always `/tmp`.

Reply:
Ah, but that's a JVMism ;) I agree with the sentiment though, the fs2.io.files method for temporary files/directories would be the best way to go about this.

Reply:
Does that do something that honors `$TMPDIR` on JS?  Regardless, I like the idea of outsourcing that portability concern to fs2 if they've already solved it.

Reply:
That's a good question. fs2 just delegates to Node.js `os.tmpdir()` method which "Returns the operating system's default directory for temporary files" whatever that means.